applications/StructuralMechanicsApplication/tests/response_function_tests/FD_strain_energy_response_parameters.json

```diff
@@ -0,0 +1,22 @@
+{
```

Comment:
@MFusseder, @armingeiser 

A more general thing here: I do like this idea of having a generic way to realize global finite differencing with any response function.  However, the present implementation comes with some implication that I would like to raise and discuss:

1. The resulting distinction between response_settings and sub_response_settings leads to a very confusing and in my opinion bad settings layout (in any case not readable by a non-expert involved in the implementation). E.g. why would there be two settings for response_type on different levels, whereas with the above one specifies the type of gradient analysis and with the lower one defines the actual physics / response output? In this context, why would one of the most important settings (i.e. the type of response function, here "strain_energy") be hidden in a sublevel at the end of the settings? I feel this should be at a much more prominent place. Thinking of a end-user I would first want to choose the physics of the response function and then the way the gradients are computed. Also, why do I have to specify some semi_analytic gradient_mode if I want to do global finite differencing in the first place?

2. Finite differencing (and adjoint sensitivity analysis) only concerns the gradient analysis, not the calculation of the value. Hence, it is just one of two parts that make up a response. However, in the current code design, one chooses different response functions (by adjusting the name of the response_type) to do adjoint analysis or FD. In contrast to that, one needs to adjust the parameter "gradient_mode" to do semi-analytic sensitivity analysis. I know there is a lot of details and difficulty in the implementation, but from a dumb user perspective, shouldn't this be specified by a single parameter, say "gradient_analysis_type" or "gradient_mode", on ONE settings level. 

While I understand the way it is implemented, I feel it is very hard for a new person to understand the overall process. And in any case it requires a lot of insight and adjustments to define a set responses for an application somewhere. 

Is there a way we can solve this issue? What do you think?

Reply:
I definitely agree. The current settings setup reflects the implementation, but is not user friendly.
Especially because the gradient related settings of the "sub_response" are unused...

I would also prefer reuse the gradient mode setting, in that case i think the response function factory needs to get a little more logic. But i think it could be done. Maybe we can discuss it in person to agree on some details before we put effort of implementing?